analyzer/validatorbalances/validatorbalances.go

```diff
@@ -0,0 +1,109 @@
+package evmtokens
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/oasisprotocol/nexus/analyzer"
+	"github.com/oasisprotocol/nexus/analyzer/item"
+	"github.com/oasisprotocol/nexus/analyzer/queries"
+	"github.com/oasisprotocol/nexus/config"
+	staking "github.com/oasisprotocol/nexus/coreapi/v22.2.11/staking/api"
+	"github.com/oasisprotocol/nexus/log"
+	"github.com/oasisprotocol/nexus/storage"
+	"github.com/oasisprotocol/nexus/storage/oasis/nodeapi"
+)
+
+// The validator balance analyzer (1) gets a list of epochs to download validator
+// balance info for, (2) downloads that info, and (3) saves the info in the database.
+
+const (
+	//nolint:gosec // thinks this is a hardcoded credential
+	validatorBalanceAnalyzerName = "validator_balance_"
+)
+
+type processor struct {
+	source nodeapi.ConsensusApiLite
+	target storage.TargetStorage
+	logger *log.Logger
+}
+
+var _ item.ItemProcessor[*Epoch] = (*processor)(nil)
+
+func NewAnalyzer(
+	cfg config.ItemBasedAnalyzerConfig,
+	sourceClient nodeapi.ConsensusApiLite,
+	target storage.TargetStorage,
+	logger *log.Logger,
+) (analyzer.Analyzer, error) {
+	logger = logger.With("analyzer", validatorBalanceAnalyzerName)
+	p := &processor{
+		source: sourceClient,
+		target: target,
+		logger: logger,
+	}
+
+	return item.NewAnalyzer[*Epoch](
+		validatorBalanceAnalyzerName,
+		cfg,
+		p,
+		target,
+		logger,
+	)
+}
+
+type Epoch struct {
```

Comment:
Could this be made private?

e.g.
```suggestion
type epoch struct {
```

(and also the fields)

Reply:
yep, updated!